WordPress/src/main/java/org/wordpress/android/ui/jetpack/scan/ScanViewModel.kt

```diff
@@ -243,16 +246,20 @@ class ScanViewModel @Inject constructor(
                     }
                     if (!status.containsOnlyErrors) {
                         someOrAllThreatFixed = true
-                    } else if (isInvokedByUser) {
+                    } else {
                         messageRes = R.string.threat_fix_all_status_error_message
                     }
                 }
             }
-            updateUiState(
-                    buildContentUiState(model = requireNotNull(scanStateModel), fixingThreatIds = fixingThreatIds)
-            )
 
-            messageRes?.let { updateSnackbarMessageEvent(UiStringRes(it)) }
+            val shouldUpdateUi = status is FetchFixThreatsState.InProgress || status is FetchFixThreatsState.Complete
+            if (shouldUpdateUi) {
```

Comment:
Suggestion (💡): This logical change occurred as part of this eab11fa33683a22f67afc2f07e5db57da4fa62e3 commit. The commit message reads as:

```
When checking fix status, update content only for in progress and complete fix states
```

When reading this commit, although I understand the `what`, I am missing context on the `why`.

In cases like that, I usually prefer to add a description to my commit message explaining to the reader the `why` of that change. I do that because I believe that this will make it easier for the reviewer to understand the change and more so it will make it easier for any engineer that will stumble upon that change later on, with git blame or otherwise, to understand its necessity.

PS: I always try to be as much descriptive on my commits as I really believe that a code is written once, but on average it is being read then many more times. Thus, IMHO, an extra investment it is always worth it.
PSS: Please take this comment with a grain of salt since I am indeed biased with this methodology, been using it for years and got pretty comfortable with it and the extra effort it requires of me.

Reply:
You're right, it could have a better description. Let me try to explain the "Why" below so that we have the context for the future:

**Reason 1**: Earlier the content UI was updated for all fix statuses, however, it is not necessary to update for the remaining states:

- `NotStarted`:  No change in the existing threats state.
- `Failure.NetworkUnavailable,` `Failure.RemoteRequestFailure`:  Request didn't complete so we could not determine any changes in the existing threats state.
- `Failure FixFailure`: If none of the threats could be fixed, there's no change in the existing threats state. If some of the threats are fixed, scan state is re-fetched [here](https://github.com/wordpress-mobile/WordPress-Android/blob/develop/WordPress/src/main/java/org/wordpress/android/ui/jetpack/scan/ScanViewModel.kt#L195) which displays the updated threats state. 

**Reason 2**: If called from the VM init, the scan button was earlier shown during an unnecessary update on states like `Not Started `and other `Failure` states. An early display of the scan button runs the risk of early scan start by the user (before fetch scan state request in VM start completes) causing the issue we're tacking in this PR.

Hope this helps and makes sense 🙂.

> Please take this comment with a grain of salt 

No worries, in fact, thank you for all the thoughtful comments 🙏.

Reply:
Thank you so much for all this detailed explanation @ashiagr , you rock! 🙏  🥇